lib/plugins/threaded-signing/plugin.c

```diff
@@ -123,17 +123,22 @@ sv_threaded_plugin_reset(sv_threaded_plugin_t *self)
   local_signature_info_free(self->signature_info);
   self->signature_info = NULL;
 
-  for (int i = 0; i < MAX_BUFFER_LENGTH; i++) {
-    free(self->output_buffer[i].signature);
-    self->output_buffer[i].signature = NULL;
-    self->output_buffer[i].size = 0;
-    self->output_buffer[i].signing_error = false;
+  int j = 0;
+  while (self->output_buffer[j].signature != NULL) {
```

Comment:
I think you should keep the for loop and free _all_ elements. If there by accident has become a gap in the buffer, that is, on NULL pointer has sneaked in in the middle of true pointers, you will not free all of the memory.
There is no need to minimize operations in the reset call, since it is only done when something has gone wrong or when tearing down the session.